src/DF950.c

```diff
@@ -76,9 +231,36 @@ void sfx_reset_door_sounds(void) {
     D_80151308 = 0;
 }
 
-INCLUDE_ASM(void, "DF950", sfx_clear_sounds);
+void sfx_clear_sounds(void) {
+    bzero(&D_801598A0, sizeof(D_801598A0));
+    bzero(&D_801599B8, sizeof(D_801599B8));
+    D_80159AD0 = D_801598A0;
+    func_801497FC(0);
+    sfx_reset_door_sounds();
+}
+
+// name might be incorrect?
+void sfx_clear_env_sounds(s16 playSounds) {
+    if (!gGameStatusPtr->isBattle) {
+        D_80159AD0 = D_801598A0;
+    } else {
+        D_80159AD0 = D_801599B8;
+    }
+
+    if (playSounds) {
+        s32 i;
+        SoundInstance* sound = D_80159AD0;
+        for (i = 0; i < MAX_SOUND_INSTANCES; sound++) {
+            if (sound->flags & 1) {
+                snd_start_sound_with_shift(sound->soundID, sound->volume, sound->pan, sound->pitchShift);
+            }
 
-INCLUDE_ASM(s32, "DF950", sfx_clear_env_sounds);
+            i++;
```

Comment:
Does it not work to do `i++, sound++` in the loop definition? Also I assume D_80159AD0[i] doesn't work and you actually need to increment the sound temp? Not ideal but not the first time we've seen it